src/components/playback/mediasession.js

```diff
@@ -49,122 +32,101 @@ define(['playbackManager', 'nowPlayingHelper', 'events', 'connectionManager'], f
-        pushImageUrl(item, {height: 384}, list);
-        pushImageUrl(item, {height: 512}, list);
+        imageSizes.forEach((size) => {
+            const url = getImageUrl(item, {height: size});
+            if (url !== null) {
+                list.push(url);
+            }
+        });
 
         return list;
     }
 
     function updatePlayerState(player, state, eventName) {
         // Don't go crazy reporting position changes
-        if (eventName == 'timeupdate') {
+        if (eventName === 'timeupdate') {
             // Only report if this item hasn't been reported yet, or if there's an actual playback change.
             // Don't report on simple time updates
             return;
         }
 
-        var item = state.NowPlayingItem;
+        const item = state.NowPlayingItem;
 
         if (!item) {
             hideMediaControls();
             return;
         }
 
-        if (eventName == 'init') { // transform "init" event into "timeupdate" to restraint update rate
+        if (eventName === 'init') { // transform "init" event into "timeupdate" to restraint update rate
             eventName = 'timeupdate';
         }
 
-        var isVideo = item.MediaType === 'Video';
-        var isLocalPlayer = player.isLocalPlayer || false;
+        const isVideo = item.MediaType === 'Video';
+        const isLocalPlayer = player.isLocalPlayer || false;
 
         // Local players do their own notifications
         if (isLocalPlayer && isVideo) {
             return;
         }
 
-        var playState = state.PlayState || {};
-        var parts = nowPlayingHelper.getNowPlayingNames(item);
-        var artist = parts[parts.length - 1].text;
-        var title = parts.length === 1 ? '' : parts[0].text;
-        var albumArtist;
-
-        if (item.AlbumArtists && item.AlbumArtists[0]) {
-            albumArtist = item.AlbumArtists[0].Name;
-        }
+        const playState = state.PlayState || {};
+        const parts = nowPlayingHelper.getNowPlayingNames(item);
+        const artist = parts[parts.length - 1].text;
+        const title = parts.length === 1 ? '' : parts[0].text;
 
-        var album = item.Album || '';
-        var itemId = item.Id;
+        const album = item.Album || '';
+        const itemId = item.Id;
 
         // Convert to ms
-        var duration = parseInt(item.RunTimeTicks ? (item.RunTimeTicks / 10000) : 0);
-        var currentTime = parseInt(playState.PositionTicks ? (playState.PositionTicks / 10000) : 0);
+        const duration = parseInt(item.RunTimeTicks ? (item.RunTimeTicks / 10000) : 0);
+        const currentTime = parseInt(playState.PositionTicks ? (playState.PositionTicks / 10000) : 0);
 
-        var isPaused = playState.IsPaused || false;
-        var canSeek = playState.CanSeek || false;
+        const isPaused = playState.IsPaused || false;
+        const canSeek = playState.CanSeek || false;
 
-        if (navigator.mediaSession) {
+        if ('mediaSession' in navigator) {
```

Comment:
I would remove all the checking here and just leave the one done at the beginning

Reply:
They have different purposes.

The one at the beginning is to check if either the Media Session API or Native Shell is available.  
This one is specifically to check if the Media Session API is available, since a client without Media Session but with Native Shell will load this file.